src/gtk/filedlg.cpp

```diff
@@ -151,6 +151,20 @@ static void gtk_filedialog_update_preview_callback(GtkFileChooser *chooser,
 
 } // extern "C"
 
+extern "C"
+{
+static gboolean gtk_frame_map_callback( GtkFileChooser *file_chooser,
+                        GdkEvent * WXUNUSED(event),
+                        wxFileDialog *dlg )
+{
+    if( dlg->GetWindowStyle() & wxFD_SHOW_HIDDEN )
+        gtk_file_chooser_set_show_hidden( file_chooser, TRUE );
+    else
```

Comment:
We don't need the `else` branch, this is the default anyhow.

Reply:
@vadz ,
Not according to my testing.
Building the dialog sample and trying "File operations -> Open file" displays everything without the else.
I have Gentoo Linux with GTK+3.22.30.

Reply:
Very strange, the docs [explicitly say](https://developer.gnome.org/gtk3/stable/GtkFileChooser.html#GtkFileChooser--show-hidden) that this is off by default.

Anyhow, if we do need to always call it, then it'd make sense to write
```
gtk_file_chooser_set_show_hidden(file_chooser, dlg->HasFlag(wxFD_SHOW_HIDDEN));
```

Reply:
@vadz ,
I don't know - maybe it just Gentoo patching GTK.

Anyway I fixed it as suggested. Please apply and close the ticket after the build passes.

TIA!